src/portfft/common/workgroup.hpp

```diff
@@ -227,9 +226,13 @@ __attribute__((always_inline)) inline void dimension_dft(
         }
       }
     }
-
-    sg_dft<Dir, SubgroupSize>(priv, global_data.sg, fact_wi, fact_sg, loc_twiddles, wi_private_scratch);
-
+    if (take_conjugate_on_load == detail::complex_conjugate::TAKEN) {
+      take_conjugate_inplace(priv, fact_wi);
+    }
```

Comment:
I'm personally not too bothered by the 2 new parameters. It seems to make the global implementation simpler and it's nice not having to give the direction parameter to the functions that compute the DFT.

Reply:
Its not strictly an optimization. On almost all hardware, `-1 * floating_point_value`,  is not a multiplication, but rather it gets optimized away. For example on CPU it would turn to an `xor` op, cuda uses a `neg.s32` ptx instruction, and these are 1 cycle ops. Given this will always be in registers,  there should not be any noticeable  difference between multiplying within the kernel and explicitly creating twiddles for the backward pass.

The main reason to change the way the backward pass works is because inverse Bluestein' FFTs will make use of this logic, so it would be required then, and also it would be nice to have all our backward passes follow the same logic

Reply:
OK, we agree on the optimization part. 

Does Bluestein need to conjugate both input and output? I think conjugating input or output might be the same operation due to linearity property of the FFT.

Or am I misunderstanding and the new implementation allows you to do something else that the previous one does not that is needed for Bluestein?

> It seems to make the global implementation simpler

I do not see any changes to global implementation or global dispatcher that would really change the complexity of the implementation.

> it's nice not having to give the direction parameter to the functions that compute the DFT.

If that is nice I would says it is doubly so not nice having to pass two parameters.

Reply:
Sorry I should reword my comment. I meant that the direction parameter is not needed anymore in most of the implementation files (global.hpp, subgroup.hpp and workitem.hpp). It is contained to the dispatcher files only, except for workgroup.hpp.
So just in terms of complexity the 2 versions look roughly equivalent to me.

Reply:
After some discussion we figured out that this is important for future Bluestein work, which is easier to implement if we have an option to conjugate just the input or just the output, compared to inverse transform that needs both conjugated. Resolving.